server/src/test/java/org/elasticsearch/cluster/service/MasterServiceTests.java

```diff
@@ -2121,6 +2121,56 @@ public void onFailure(Exception e) {
         }
     }
 
+    public void testTimeoutRejectionBehaviourAtSubmission() {
+
+        final var actionCount = new AtomicInteger();
+
+        final var deterministicTaskQueue = new DeterministicTaskQueue();
+        final var threadPool = deterministicTaskQueue.getThreadPool(r -> {
+            if (r.toString().startsWith("master service timeout handler for [batched][test task]")) {
+                actionCount.incrementAndGet();
+                throw new EsRejectedExecutionException("simulated rejection", true);
+            } else {
+                return r;
+            }
+        });
+
+        try (var masterService = createMasterService(true, null, threadPool, new StoppableExecutorServiceWrapper(threadPool.generic()))) {
+
+            class TestTask implements ClusterStateTaskListener {
```

Comment:
A descriptive name?

TestTask => ClusterStateTaskThatCountsSimulatedErrors

Reply:
Actually we can just inline this, this was more useful in the test from which I copied it. Simplest to just have the implementation right there rather than indirect via a name

Reply:
A good name can be very informative :) But that works, too 👍